docs/examples/custom-workflow.ipynb

```diff
@@ -0,0 +1,709 @@
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Template Picker\n",
+    "\n",
+    "Create and run a Template Picker job with all micrographs."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'completed'"
+      ]
+     },
+     "execution_count": 14,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "template_picker_job = workspace.create_job(\n",
+    "    \"template_picker_gpu\",\n",
+    "    connections={\n",
+    "        \"micrographs\": (ctf_estimation_job.uid, \"exposures\"),\n",
+    "        \"templates\": (select_blob_templates_job.uid, \"templates_selected\"),\n",
+    "    },\n",
+    "    params={\"diameter\": 200},\n",
+    ")\n",
+    "template_picker_job.queue(lane)\n",
+    "template_picker_job.wait_for_done()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Repeat all previous steps from Inspect Picks to Select 2D, using the template picks as input."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'completed'"
+      ]
+     },
+     "execution_count": 15,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Create and connect jobs\n",
+    "inspect_template_picks_job = workspace.create_job(\n",
+    "    \"inspect_picks_v2\",\n",
+    "    connections={\n",
+    "        \"micrographs\": (template_picker_job.uid, \"micrographs\"),\n",
+    "        \"particles\": (template_picker_job.uid, \"particles\"),\n",
+    "    },\n",
+    ")\n",
+    "\n",
+    "extract_template_picks_job = workspace.create_job(\n",
+    "    \"extract_micrographs_cpu_parallel\",\n",
+    "    connections={\n",
+    "        \"micrographs\": (inspect_template_picks_job.uid, \"micrographs\"),\n",
+    "        \"particles\": (inspect_template_picks_job.uid, \"particles\"),\n",
+    "    },\n",
+    "    params={\"box_size_pix\": 448},\n",
+    ")\n",
+    "\n",
+    "classify_template_picks_job = workspace.create_job(\n",
+    "    \"class_2D\", connections={\"particles\": (extract_template_picks_job.uid, \"particles\")}, params={\"class2D_K\": 50}\n",
+    ")\n",
+    "\n",
+    "select_templates_job = workspace.create_job(\n",
+    "    \"select_2D\",\n",
+    "    connections={\n",
+    "        \"particles\": (classify_template_picks_job.uid, \"particles\"),\n",
+    "        \"templates\": (classify_template_picks_job.uid, \"class_averages\"),\n",
+    "    },\n",
+    ")\n",
+    "\n",
+    "# Queue Jobs\n",
+    "inspect_template_picks_job.queue(lane)\n",
+    "extract_template_picks_job.queue(lane)\n",
+    "classify_template_picks_job.queue(lane)\n",
+    "select_templates_job.queue(lane)\n",
```

Comment:
extract and classify need to be waited to be "done" before continuing to refinement right?

Reply:
Yes, I'm letting the CryoSPARC scheduler take care of this in this example.